### PR TITLE
Made tests not disgusting, added special file creator and mkdir

### PR DESCRIPTION
**Pull request by jesings**

### CODE REVIEW

File: src/tests/fs_tests.rs [hunk 1]
```diff
@@ -0,0 +1,67 @@
+#[cfg(test)]
```
Comment (rennergade) [inline src/tests/fs_tests.rs:1]: Agree that moving these tests here makes sense

File: src/safeposix/filesystem.rs [hunk 2]
```diff
@@ -70,21 +70,53 @@ pub struct FilesystemMetadata {
     pub inodetable: interface::RustHashMap<usize, Inode>,
 }
 
+pub fn init_filename_to_inode_dict(parentinode: usize) -> interface::RustHashMap<std::ffi::OsString, usize> {
+    let mut retval = interface::RustHashMap::new();
+    retval.insert(std::ffi::OsString::from("."), ROOTDIRECTORYINODE);
```
Comment (rennergade) [inline src/safeposix/filesystem.rs:75]: this is a non-interface call in main safeposix code *wags finger*

File: src/safeposix/filesystem.rs [hunk 3]
```diff
@@ -70,21 +70,53 @@ pub struct FilesystemMetadata {
...
+        let dirinode = DirectoryInode {size: 0, uid: DEFAULT_UID, gid: DEFAULT_GID,
+            mode: S_IFDIR as u32 | S_IRWXA, linkcount: 3, refcount: 0, 
+            atime: time, ctime: time, mtime: time,
+            filename_to_inode_dict: init_filename_to_inode_dict(ROOTDIRECTORYINODE)};
+        retval.inodetable.insert(ROOTDIRECTORYINODE, Inode::Dir(dirinode));
         retval
     }
 }
 
+pub fn load_fs_special_files() {
```
Comment (rennergade) [inline src/safeposix/filesystem.rs:94]: Is this where we're setting up the initial cage now?

**Review by rennergade — COMMENTED** [review]
Mostly good. Had a question about how we're going to do init, and flagged some non-interface calls.

File: src/safeposix/filesystem.rs (hunk 3 above)
Comment (jesings) [reply src/safeposix/filesystem.rs:94]: No I haven't really settled on how/if the initial cage is going to work

Comment (rennergade) [reply src/safeposix/filesystem.rs:94]: OK this is fine for now, we'll probably move this around once I do my PR.

**Pull request merged**